### PR TITLE
Rewrite regenerate-templates.sh script + add patches

### PR DESCRIPTION
**Pull request by k-wall**
Resulting templates tested with latest Red Hat AMQ Online container images

### CODE REVIEW

File: regenerate-templates.sh [hunk 1]
```diff
@@ -1,9 +1,98 @@
...
+#!/usr/bin/env bash
+
+do_patch () {
+    PATCH_DIR=$1
+    PATCH_TARGET=$2
+    if [ -d "${PATCH_DIR}" ]
+    then
+        for patch in $(find ${PATCH_DIR} -type f -name "*.patch"); do
+            echo Applying patch ${patch}
+            patch --remove-empty-files --merge -f -d "${PATCH_TARGET}" -p1 < $patch
```
Comment (lulf) [inline regenerate-templates.sh:10]: Should it perhaps fail if patch did not apply ?

**Review by lulf — COMMENTED** [review]
One comment, LGTM. Some of the patches will break with next release, but hopefully not that much. 

Also, I think we should consider a single ‘amqonline’ bundle downstream, but we can do that another time.

File: regenerate-templates.sh (hunk 1 above)
Comment (lulf) [reply regenerate-templates.sh:10]: One alternative is to add ‘set -e’ in top of file to always fail on sub command errors

**Comment by k-wall** [comment]
Retested with amq-interconnect/amq-interconnect-1.2-openshift:1.0-6

File: templates/ansible/playbooks/openshift/install/address-space-controller/020-ConfigMap-address-space-definitions.yaml [hunk 2]
```diff
@@ -849,7 +849,7 @@ data:
                 value: "INFER"
               - name: QDROUTERD_AUTO_MESH_SERVICE_NAME
                 value: "qdrouterd-headless-${INFRA_UUID}"
-              image: docker.io/enmasseproject/router:0.23.0
+              image: brew-pulp-docker01.web.prod.ext.phx2.redhat.com:8888/amq-interconnect/amq-interconnect-1.2-openshift:1.0-6
```
Comment (lulf) [inline templates/ansible/playbooks/openshift/install/address-space-controller/020-ConfigMap-address-space-definitions.yaml]: I think this should be just 'amq-interconnect/amq-interconnect-1.2...' and not include internal red hat registry urls.

**Review by lulf — DISMISSED** [review]

**Review by lulf — APPROVED** [review]

**Pull request merged**